apps/3d_rec_framework/CMakeLists.txt

```diff
@@ -13,7 +13,7 @@ else(NOT VTK_FOUND)
 endif(NOT VTK_FOUND)
 
 # OpenNI found?
-if(NOT OPENNI_FOUND OR NOT BUILD_OPENNI)
+if(NOT OPENNI_FOUND OR NOT WITH_OPENNI)
```

Comment:
But consider the following usecase. The user runs `ccmake`, on the first run all the libraries that are present will be found (because `WITH` options are on by default). Then the user looks through the offered options, and decides to disable some of the `WITH` options. In this case the library is found, but should not be used.